Add footnote about actor/keyId resolution during HTTP Signature processing

mjankowski opened the pull request:
Resolves https://github.com/mastodon/documentation/issues/1431

This page was actually improved quite a bit since the original issue, but I think the nuance around this one detail is still missing. This is essentially a verbatim copy/paste from the issue into what I think is a relevant portion of the spec description/guide.

I contemplated whether this was most appropriate on this security page or the AP spec page, and opted for here. Don't feel strongly though, could move.

andypiper commented:
Requesting an additional review from @oneiros or @ClearlyClaire just to double-check here, thanks.

mjankowski commented:
I agree it's quite dense ... and admittedly I started by taking the issue wording for granted and then checking code to verify, rather than the other way around. I wonder if one/both of...

- Use bullets to keep the actor-fragment case and key case more separated?
- Add more of the verbiage from the matrix chat in original issue to expand even more on the logic?

...would help

mjankowski commented:
Updated a bit to both include a little more leading/intro text in the footnote, and split to bullets for the main paths.

I think this makes it marginally more clear, but its still pretty terminology-dense and if you are not already deep in the weeds on this its probably nonsensical.
